### DISCUSSION TITLE
Templates are rendered as plain text, not HTML

### DISCUSSION THREAD

**Discussion by s0lst1ce** [rwf2/Rocket#2238] [Questions] [answered]
Hello! I'm failing to render templates with rocket, all I get is the plain text of my templates file. Why is that? Here's my handlebars file (called `templates/index.hbs`) 
```hbs
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">

    <title>Document</title>
</head>

<body>
    <h2>Filtres & Zapier</h2>
    <form action="/">
        <h3>Filtres scrapping</h3>
        <label for="deps">Departements (ex: "34,75"):</label><br>
        <input type="text" id="deps" name="deps" , value={{deps}}><br>
        <label for="type">Type ("pro", "particulier" ou vide):</label><br>
        <input type="text" id="type" name="type"><br><br>
        <label for="category">Catégorie ("vente", "location" ou vide):</label>
        <input type="text" id="category" name="category"><br><br>
        <h3>Paramètres Zapier</h3>
        <label for="url">URL du webhook Zapier:</label>
        <input type="text" id="url" name="url"><br><br>
        <label for="send_start">Début de l'envoie (ex: "1000" pour 10h00)</label>
        <input type="text" id="send_start" name="send_start"><br><br>
        <label for="send_end">Fin de l'envoie (ex: "1700" pour 17h00)</label>
        <input type="text" id="send_end" name="send_end"><br><br>
        <label for="intersperse">Interval entre l'envoie de deux annonces à Zapier, en secondes.</label>
        <input type="text" id="intersperse" name="intersperse"><br><br>

        <input type="submit" value="Submit">
    </form>
    <br>

</body>

</html>
```
And my route that's supposed to display it: 
```rs
#[get("/")]
async fn show(backend: &State<TomlBackend>) -> Result<Template, AppError> {
    let context: Context = backend.inner().lock().await.get().await?.into();
    Ok(Template::render("index", context))
}
```
But all I see is this, which is mot certainly not what I want
![image](https://user-images.githubusercontent.com/46009999/174588478-a4ed050a-621b-4923-9e7b-ba615f57aa20.png)
Actually I see the same issue when returning a String: 
```rs
#[get("/test")]
async fn test() -> String {
    String::from("<h2>Hello!</h2>")
}
```
I remember there being an `Html` responder but I can't find it anymore in rocket 0.5.

**Comment by s0lst1ce** [comment]
And here's my entrypoint:
```rs
#[launch]
async fn rocket() -> _ {
    dotenv::dotenv().unwrap();
    let path = env::var("TOML_SETTINGS_PATH")
        .expect("Couldn't find env var `TOML_SETTINGS_PATH`, make sure it is set correctly.");
    let backend = Mutex::new(recommended_backend(path).await);
    rocket::build()
        .mount("/", routes![show, update, test])
        .manage(backend)
        .attach(Template::fairing())
}
```

**Comment by SergioBenitez** [comment, marked as answer]
Templates don't need to be HTML, so there's no way for Rocket to know what kind of content they contain unless you specify it via a file extension. That is, naming your template something like `index.html.hbs` instead of just `index.hbs`. This is specified in the [docs](https://api.rocket.rs/v0.5-rc/rocket_dyn_templates/index.html#content-type):

> The Content-Type of the response is automatically determined by the non-engine extension of the template name or text/plain if there is no extension or the extension is unknown. For example, for a discovered template with file name foo.html.hbs or a manually registered template with name ending in foo.html, the Content-Type is automatically set to ContentType::HTML.

**Reply by s0lst1ce** [reply]
Many thanks, I hadn't understood that! I though the doctype was enough.
